### PR TITLE
Use correct type for as and forwardedAs props

### PR DESCRIPTION
**Pull request by jonkoops**

### CODE REVIEW

File: packages/asc-ui/src/components/Typography/TypographyStyle.ts [hunk 1]
```diff
@@ -24,11 +25,16 @@ export const defaultTypographyStyles = {
...
 
-const getProperty = <T, K extends keyof T>(obj: T, key: K) => obj[key]
-
 export default styled.p<Props>`
-  ${({ as, forwardedAs }) =>
-    getProperty(defaultTypographyStyles, as || forwardedAs)};
+  ${({ as, forwardedAs }) => {
+    const key = as ?? forwardedAs
+
+    if (typeof key === 'string') {
```
Comment (robinpiets) [inline packages/asc-ui/src/components/Typography/TypographyStyle.ts:32]: Does this work?
`return typeof key === 'string' && defaultTypographyStyles[key] ?? null`?

Comment (jonkoops) [reply packages/asc-ui/src/components/Typography/TypographyStyle.ts]: Actually, the function can return any falsy value so we can even drop the `null`. Changed it.

File: packages/asc-ui/src/components/Typography/TypographyStyle.ts [hunk 2]
```diff
@@ -7,28 +8,31 @@ export type Props = {
   element?: Variant
   fontSize?: number
   styleAs?: keyof Theme.TypographyElements
```
Comment (robinpiets) [inline packages/asc-ui/src/components/Typography/TypographyStyle.ts:10]: Can this be `ElementType` as well?

Comment (jonkoops) [reply packages/asc-ui/src/components/Typography/TypographyStyle.ts:10]: No we want to allow a specific subset of elements here that can only be used in typography,

**Review by robinpiets — APPROVED** [review]

**Pull request merged**